server/src/main/java/org/candlepin/controller/CandlepinPoolManager.java

```diff
@@ -1597,105 +1624,30 @@ public Entitlement adjustEntitlementQuantity(Consumer consumer, Entitlement enti
         Map<String, Integer> poolQuantityMap, CallerType caller)
         throws EntitlementRefusedException {
 
-        // Because there are several paths to this one place where entitlements
-        // are granted, we cannot be positive the caller obtained a lock on the
-        // pool when it was read. As such we're going to reload it with a lock
-        // before starting this process.
-        log.debug("Locking pools: {}", poolQuantityMap.keySet());
-        Collection<Pool> pools = poolCurator.lockAndLoadByIds(poolQuantityMap.keySet());
-
-        if (log.isDebugEnabled()) {
-            for (Pool pool : pools) {
-                log.debug("Locked pool: {} consumed: {}", pool, pool.getConsumed());
-            }
-        }
-
-        log.debug("Done locking pools");
+        BindChain chain = new BindChain();
+        BindContext context = bindContextFactory.create(consumer, poolQuantityMap);
+        chain.addOperation(rulesCheckOpFactory.create(caller));
+        chain.addOperation(handleEntitlementsOp);
+        chain.addOperation(postBindBonusPoolsOp);
+        chain.addOperation(handleCertificatesOp);
+        chain.addOperation(complianceOp);
 
```

Comment:
Yep, no worries. I'm interested in @awood comments.

I don't think that this change strays from the design. The change is simply hiding details of execution and the execution context from the caller.

A few other comments... If the overall intent is to be able to create a generic chain of varying operations, and then to be able to reuse 'the concept' in other places, then we shouldn't be calling it a "BindChain" and typing the objects it uses as such. As it is right now, it is defining the exact steps required in the bind process.